<launch>
    <arg name="cmd_package" default="turtlesim" doc="package of the node publissinh in cmd_input"/>
    <arg name="cmd_node" default="turtle_teleop_key"/>
    <arg name="cmd_output" default="turtle1/cmd_vel"/>
    <arg name="name" default="command"/>
    <node pkg="turtlesim" name="sim" type="turtlesim_node"/>
    
    <node pkg="safe_turtles" name="muxer" type="mux">
        <remap from="output" to="/turtle1/cmd_vel"/>
    </node> 

    <node pkg="$(arg cmd_package)" name="$(arg name)" type="$(arg cmd_node)">
        <remap from="$(arg cmd_output)" to="cmd_input"/>
    </node>

    <node pkg="safe_turtles" name="st_node" type="safe_turtles_node">
        <remap from="input" to="/turtle1/pose"/>
        <remap from="output" to="/safe_input"/>
    </node> 
</launch>
